netbox/dcim/choices.py

```diff
@@ -1151,6 +1176,7 @@ class PortTypeChoices(ChoiceSet):
     TYPE_URM_P4 = 'urm-p4'
     TYPE_URM_P8 = 'urm-p8'
     TYPE_OTHER = 'other'
+    TYPE_BACKPLANE = 'backplane'
```

Comment:
The scope of #10356 is limited to the introduction of new interface types. Any new port/cable types would need to be proposed separately.

Reply:
Fair enough. I forgot to include the new port type initially and only were missing them when testing and modeling my requirements. I'll create a new ticket for that.
But the new cable type is included in the ticket description.

Reply:
Sorry, I missed the cable type. Would "stacking" maybe be a better name for this? I've just never heard the term "backplane cable;" open to other opinions.

Reply:
I removed the new port type and will create a followup ticket to have that added.

A backplane connection is no stacking (like in virtual-chassis, etc). It is merely a direct physical connection without any cable :/ Nevertheless the cable description was adjusted to hopefully better describe it. If you come up with a better/more appropriate name feel free to change that. As said there actually  is no cable in use, when a backplane connection is established. It's just a shortcoming of the connection/cable model in netbox and I'm not questioning that nor do I propose any changes. I'd just like to have a way to filter and search for backplane connections/cables, so it needs an identifier. The type "other" wouldn't match reality too, imho.

Reply:
What's the difference between a backplane connection and a module? For something like modular Cisco chassis (Nexus 7700 and Catalyst 9500), they just use slots that slide into the backplace, but I'm curious as to the reasoning for modeling the backplane?

Reply:
Imho a module is more or less limited to networking equipment (at least that's what it's designed for). My use case comes from modular server systems e.g. bladecenters. You don't want to model a blade as a module. A blade is a fully fledged child device that goes into device-bays of a chassis.

E.g. https://www.supermicro.com/en/products/superblade/module/SBI-4129P-C2N.cfm
This blade has 2x 10G interfaces that are backplane connections to the switch/passthru/whatever is plugged into the bladecenter chassis.

An ethtool output of one interface on those blades show:

```
user# ethtool eno1
Settings for eno1:
	Supported ports: [ Backplane ]
	Supported link modes:   1000baseKX/Full
	                        10000baseKR/Full
	Supported pause frame use: Symmetric Receive-only
	Supports auto-negotiation: Yes
	Supported FEC modes: Not reported
	Advertised link modes:  1000baseKX/Full
	                        10000baseKR/Full
	Advertised pause frame use: No
	Advertised auto-negotiation: Yes
	Advertised FEC modes: Not reported
	Speed: 10000Mb/s
	Duplex: Full
	Auto-negotiation: on
	Port: None
	PHYAD: 0
	Transceiver: internal
	Supports Wake-on: g
	Wake-on: g
        Current message level: 0x00000007 (7)
                               drv probe link
	Link detected: yes
```

This whole PR is for modeling those kind of devices.

Reply:
> A blade is a fully fledged child device that goes into device-bays of a chassis.

These are modeled in NetBox as child devices. There is no need to model a discrete connection as it's implied by the parent/child relationship.

I think we should omit the "backplane" connection type as it's too ambiguous, and easily confused with e.g. passive direct attach cabling.